src/Parser.js

```diff
@@ -591,62 +592,63 @@ class Parser {
 
     /**
      * Call a function handler with a suitable context and arguments.
+     * @param {string} name
+     * @param {Array<ParseNode>} args
+     * @param {Array<?ParseNode>} optArgs
+     * @param {Token=} token
```

Comment:
Is the plan to turn this into flow types at a later date?

Reply:
Yes. The point is to write down all our intended types while we refactor to make porting to flow possible. That requires the `ParseNode` refactor we discussed in #892. Until we're there, writing out these types aids keeping our intended mental model better documented.